structural/script.js

```diff
@@ -300,6 +300,295 @@ const formula_circle_I_y_bar = [
     },
 ];
 
+const formula_half_circle_x_bar = [
+    {
+        displayName: 'Calculate x_bar',
+        calculate: (half_cir_D) => half_cir_D/2
+    },
+    {
+        displayName: 'Calculate D',
+        calculate: (half_cir_x_bar) => half_cir_x_bar*2
+    },
+];
+
+const formula_half_circle_y_bar = [
+    {
+        displayName: 'Calculate y_bar',
+        calculate: (half_cir_r) => (4*half_cir_r)/(3*Math.PI)
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (half_cir_y_bar) => (3*Math.PI*half_cir_y_bar) / 4
+    },
+];
+
+const formula_half_circle_area = [
+    {
+        displayName: 'Calculate area',
+        calculate: (half_cir_r1) => (Math.PI * half_cir_r1**2) / 2
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (half_cir_area) => (2 * half_cir_area / Math.PI)**(1/2)
+    },
+];
+
+const formula_half_circle_I_x_bar = [
+    {
+        displayName: 'Calculate I_x_bar',
+        calculate: (half_cir_r2) => ((9*(Math.PI**2) - 64) / 72*Math.PI) * half_cir_r2**4
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (half_cir_I_x_bar) => ((72*Math.PI*half_cir_I_x_bar) / (9*(Math.PI**2) - 64))**(1/4)
+    },
+];
+
+const formula_half_circle_I_y_bar = [
+    {
+        displayName: 'Calculate I_y_bar',
+        calculate: (half_cir_r3) => (Math.PI * half_cir_r3**4) / 8
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (half_cir_I_y_bar) => (8 * half_cir_I_y_bar / Math.PI)**(1/4)
+    },
+];
+
+const formula_quarter_circle_x_bar = [
+    {
+        displayName: 'Calculate x_bar',
+        calculate: (quarter_cir_r) => (4*quarter_cir_r)/(3*Math.PI)
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (quarter_cir_x_bar) => (3*Math.PI*quarter_cir_x_bar) / 4
+    },
+];
+
+const formula_quarter_circle_y_bar = [
+    {
+        displayName: 'Calculate y_bar',
+        calculate: (quarter_cir_r1) => (4*quarter_cir_r1)/(3*Math.PI)
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (quarter_cir_y_bar) => (3*Math.PI*quarter_cir_y_bar) / 4
+    },
+];
+
+const formula_quarter_circle_area = [
+    {
+        displayName: 'Calculate area',
+        calculate: (quarter_cir_r2) => (Math.PI * quarter_cir_r2**2) / 4
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (quarter_cir_area) => (4 * quarter_cir_area / Math.PI)**(1/2)
+    },
+];
+
+const formula_quarter_circle_I_x_bar = [
```

Comment:
same with this one
if  I x bar = 1 then radius = 2.0660911663590564
if radius = 2.0660911663590564 then I x bar =  9.869604401089358